Documentation/UsingSetting/AddTypoScriptWithExtensions.rst

```diff
@@ -6,60 +6,167 @@
+
+.. _extdev-add-typoscript-sets-main:
+
+The main set of the extension
+-----------------------------
+
+..  literalinclude:: _Sets/_MyExtension_config.yaml
+    :language: yaml
+    :caption: EXT:my_extension/Configuration/Sets/MyExtension/config.yaml
+
+
+.. _extdev-add-typoscript-sets-feature:
+
+The sub set for an optional feature
+-----------------------------------
+
+..  literalinclude:: _Sets/_MyExtensionWithACoolFeature_config.yaml
+    :language: yaml
+    :caption: EXT:my_extension/Configuration/Sets/MyExtensionWithACoolFeature/config.yaml
+
+
+.. _extdev-add-typoscript-sets-typoscript:
+
+TypoScript files provided by the sets
+-------------------------------------
+
+The TypoScript placed in the same folder like the set, contains your
+configurations.
 
-.. index:: TypoScript in extensions; Static includes
-.. _extdev-static-includes:
 
-Make TypoScript available for static includes
-=============================================
+.. _extdev-add-typoscript-sets-v12:
 
-.. code-block:: php
-   :caption: EXT:my_extension/Configuration/TCA/Overrides/sys_template.php
+TypoScript provided by extensions supporting TYPO3 v12.4 and v13
+================================================================
 
-   <?php
-   defined('TYPO3') or die();
+When an extension provides TypoScript and should be compatible with both
+TYPO3 v12.4 and v13, you can provide site sets but still support including
+the TypoScript in the :sql:`sys_template` record via `static_file_include`'s.
 
-   call_user_func(function()
-   {
-      $extensionKey = 'myextension';
+The files in the sets are the same as in the
+:ref:`example for TYPO3 v13 only <extdev-add-typoscript-sets>`.
 
-      /**
-       * Default TypoScript
-       */
-      \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addStaticFile(
-         $extensionKey,
-         'Configuration/TypoScript',
-         'Some descriptive title'
-      );
-   });
+The extended file structure of the extension could, for example look like this:
+
+..  directory-tree::
+    :show-file-icons: true
+
+    *   Classes
+
+        *   ...
+
+    *   Configuration
+
+        *   Sets
+
+            *   MyExtension
+
+                *   :ref:`config.yaml <extdev-add-typoscript-sets-main>`
+                *   :ref:`constants.typoscript <extdev-add-typoscript-sets-typoscript>`
+                *   :ref:`setup.typoscript <extdev-add-typoscript-sets-typoscript>`
+
+            *   MyExtensionWithACoolFeature
+
+                *   :ref:`config.yaml <extdev-add-typoscript-sets-feature>`
+                *   :ref:`setup.typoscript <extdev-add-typoscript-sets-typoscript>`
+
+        *   TCA
+
+            *   Overrides
+
+                sys_template.php
+
+
+    *   Resources
+
+        *   ...
+
+    *   composer.json
+    *   ...
+
+
+.. _extdev-add-typoscript-sets-v12-static_includes:
+.. _extdev-static-includes:
+
+Only for TYPO3 v12.4: Make TypoScript available for static includes
```

Comment:
Only v12.4 is wrong, you can still go the old way with static includes (as sys_template records are still available and considered). The site sets are an additional way to configure TypoScript.

Reply:
Using static includes is not recommended anymore for TYPO3 v13.